cmd/crane/cmd/append.go

```diff
@@ -53,6 +56,22 @@ func NewCmdAppend(options *[]crane.Option) *cobra.Command {
 				return fmt.Errorf("appending %v: %v", newLayers, err)
 			}
 
+			if baseRef != "" && annotate {
+				ref, err := name.ParseReference(baseRef)
+				if err != nil {
+					return fmt.Errorf("parsing ref %q: %v", baseRef, err)
+				}
+
+				baseDigest, err := base.Digest()
+				if err != nil {
+					return err
+				}
+				img = mutate.Annotations(img, map[string]string{
+					specsv1.AnnotationBaseImageName:   ref.String(),
```

Comment:
`baseRef` might be a digest, we don't necessarily have a tag here, right?

Reply:
Yeah, what should we do in that case? (Technically `ko` has this problem too, you could do `defaultBaseImage: gcr.io/distroless/static@sha256:...`)

Given `crane append -b ubuntu@sha256:abc`, should we:
a.) just not set the `base.image.name` annotation, callers can maybe find value in the `base.image.digest` annotation by itself?
b.) not set either annotation, you don't get base image info 🤷 
c.) only set `base.image.name` to the fully-specified ref with digest, since this can still be useful for signaling base image info, even if it means rebasing/rebuilding won't be useful.

(we could maybe add this expected behavior to the spec)

Reply:
If we don't have a tag, I think we should drop the `base.image.name` annotation and just set the digest annotation.

That's annoying for situations where you want to pull by digest (e.g. `ko`), but we could hack around that a little bit. If base is specified with both tag and digest as:

```
foo:v1@sha256:abc
```

Then we could know that `name=foo:v1` and `digest=sha256:abc`.

If we lean into that convention, we could even have `ko` log a warning or something if your base image is out of date before even pushing anything.